lib/tests/2wp.js

```diff
@@ -863,6 +863,56 @@ const execute = (description, getRskHost) => {
 
     });
 
+    it('should do a pegout and round down the weis to satoshis as expected', async () => {
+
+      // Arrange
+
+      // Create a pegin for the sender to ensure there is enough funds to pegout and because this is the natural process
+      const senderRecipientInfo = await createSenderRecipientInfo(rskTxHelper, btcTxHelper);
+      const peginValueInSatoshis = btcToSatoshis(0.5);
+      const btcPeginTxHash = await sendPegin(rskTxHelper, btcTxHelper, senderRecipientInfo.btcSenderAddressInfo, satoshisToBtc(peginValueInSatoshis));
+      await ensurePeginIsRegistered(rskTxHelper, btcPeginTxHash);
+
+      const initial2wpBalances = await get2wpBalances(rskTxHelper, btcTxHelper);
+      const initialSenderAddressBalanceInSatoshis = await getBtcAddressBalanceInSatoshis(btcTxHelper, senderRecipientInfo.btcSenderAddressInfo.address);
+      const pegoutValueInRbtc = 0.04123456789123456;
+      const expectedPegoutValueInRbtc = 0.04123456; // rounded down
+
+      // Act
+
+      const pegoutTransaction = await sendTxToBridge(rskTxHelper, pegoutValueInRbtc, senderRecipientInfo.rskRecipientRskAddressInfo.address);
+
+      // Assert
+
+      let bridgeStateAfterPegoutCreation;
+
+      // Callback to get the bridge state after the pegout is created
+      const pegoutCreatedCallback = async () => {
+        bridgeStateAfterPegoutCreation = await getBridgeState(rskTxHelper.getClient());
+      };
+
+      const callbacks = {
+        pegoutCreatedCallback
+      };
+
+      await triggerRelease(rskTxHelpers, btcTxHelper, callbacks);
+
+      // Checking all the pegout events are emitted and in order
+      const blockNumberAfterPegoutRelease = await rskTxHelper.getBlockNumber();
+      const pegoutsEvents = await getPegoutEventsInBlockRange(rskTxHelper, pegoutTransaction.blockNumber, blockNumberAfterPegoutRelease);
+      await assertSuccessfulPegoutEventsAreEmitted(pegoutsEvents, pegoutTransaction.transactionHash, senderRecipientInfo, expectedPegoutValueInRbtc, bridgeStateAfterPegoutCreation);
+
+      await assert2wpBalanceAfterSuccessfulPegoutWithLargeWeis(initial2wpBalances, pegoutValueInRbtc, btcToSatoshis(expectedPegoutValueInRbtc));
+
+      // Assert that the sender address balance is increased by the actual pegout value
+      const finalSenderAddressBalanceInSatoshis = await getBtcAddressBalanceInSatoshis(btcTxHelper, senderRecipientInfo.btcSenderAddressInfo.address);
+      const releaseBtcEvent = pegoutsEvents[pegoutsEvents.length - 1];
```

Comment:
I think we can abstract this logic of getting the release event using the last index into a method, just to give a bit of context on what this means `pegoutsEvents.length - 1`. Wdyt?

Reply:
Well, the `release_btc` event is the last pegout related event emitted. I'm using indices in the tests to access the vents because this way we also check that the events are being emitted in the correct order.
I'm not using things like `Array::find`, etc., because of what I explained above and also because that would be inefficient.